packages/block-library/src/post-content/edit.js

```diff
@@ -45,40 +51,104 @@ function ReadOnlyContent( {
 	);
 }
 
-function EditableContent( { context = {} } ) {
-	const { postType, postId } = context;
+function EmptyContentPlaceholder( { context, onClose, openInserter } ) {
+	const { postType } = context;
+	const label =
+		'page' === postType
+			? __( 'This page’s content is empty' )
+			: __( 'This post’s content is empty' );
+	return (
+		<Placeholder
+			icon={ icon }
+			label={ label }
+			instructions={ __(
+				'Add your first block or pattern to get started.'
+			) }
+		>
+			<Button variant="primary" onClick={ openInserter }>
+				{ __( 'Choose a pattern' ) }
+			</Button>
+
+			<Button variant="secondary" onClick={ onClose }>
+				{ __( 'Start blank' ) }
+			</Button>
+		</Placeholder>
+	);
+}
+
+function PostContentPlaceholder( { layoutClassNames } ) {
+	const blockProps = useBlockProps( { className: layoutClassNames } );
+	return (
+		<div { ...blockProps }>
+			<Placeholder
+				className="wp-block-post-content__content-placeholder"
+				withIllustration
+			>
+				<p>
+					{ __( 'This block will be replaced with your content.' ) }
+				</p>
+			</Placeholder>
+		</div>
+	);
+}
 
+function EditableContent( { context = {}, clientId } ) {
+	const { postType, postId } = context;
+	const { selectBlock, insertBlock } = useDispatch( blockEditorStore );
 	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
 		'postType',
 		postType,
 		{ id: postId }
 	);
 
-	const entityRecord = useSelect(
-		( select ) => {
-			return select( coreStore ).getEntityRecord(
-				'postType',
-				postType,
-				postId
-			);
-		},
-		[ postType, postId ]
+	const setInserterIsOpened = useSelect(
+		( select ) =>
+			select( blockEditorStore ).getSettings()
+				.__experimentalSetIsInserterOpened,
+		[]
 	);
 
-	const hasInnerBlocks = !! entityRecord?.content?.raw || blocks?.length;
-
-	const initialInnerBlocks = [ [ 'core/paragraph' ] ];
+	const hasInnerBlocks = !! blocks?.length;
```

Comment:
I've changed this line so that the placeholder only cares whether the post content block has any child blocks, and will show even if there's saved post content.

This allows us to show the placeholder in the post editor in template preview mode as well.

Do folks see any downsides here?

Reply:
The failing E2E tests are real and I think due to this change. 

I think maybe reverting this change and then do a quick follow up to this PR to avoid a large change set.

Reply:
Reverted in [7560cba](https://github.com/WordPress/gutenberg/pull/57572/commits/7560cba8b4ec7b5ef3b5aa4d62cf3b36a7f51e08)